File: examples/symm/periodic_cell.cpp
#include<iostream>
#include<symm/cell.hpp>

int main(){
  qpp::vector3<double> a(0,2.73,2.73);
  qpp::vector3<double> b(2.73,0,2.73);
  qpp::vector3<double> c(2.73,2.73,0);

  qpp::periodic_cell<double> cl(a,b,c);
  qpp::index_range R({-2,-2,-2},{2,2,2,});
  auto i = R.__iter__();
  for (; !i.end(); i++)
    std::cout << cl.transform(qpp::vector3<double>(0),i)<< "\n";
}

File: pyexamples/periodic_cell.py
import qpp
'''
This example showshow to replicate lattice nodes using periodic_cell and qpp.index
'''
qpp.REAL = 'float'
a = qpp.vector3(0,2.73,2.73)
b = qpp.vector3(2.73,0,2.73)
c = qpp.vector3(2.73,2.73,0)

cl = qpp.periodic_cell(a,b,c)

for I in qpp.index_range([-2,-2,-2],[2,2,2]):
    print(cl.transform(qpp.vector3(0),I))
